src/Microsoft.ML.Transforms/GcnTransform.cs

```diff
@@ -225,477 +255,725 @@ public void Save(ModelSaveContext ctx)
+                verReadableCur: 0x00010003,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
                 loaderSignatureAlt: LoaderSignatureOld,
-                loaderAssemblyName: typeof(LpNormNormalizerTransform).Assembly.FullName);
+                loaderAssemblyName: typeof(LpNormalizingTransform).Assembly.FullName);
         }
 
         private const string RegistrationName = "LpNormNormalizer";
 
         // REVIEW: should this be an argument instead?
-        private const Float MinScale = (Float)1e-8;
+        private const float MinScale = (float)1e-8;
 
-        private readonly ColInfoEx[] _exes;
+        public IReadOnlyCollection<ColumnInfoBase> Columns => _columns.AsReadOnly();
+        private readonly ColumnInfoBase[] _columns;
 
-        /// <summary>
-        /// A helper method to create GlobalContrastNormalizer transform for public facing API.
-        /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="name">Name of the output column.</param>
-        /// <param name="source">Name of the column to be transformed. If this is null '<paramref name="name"/>' will be used.</param>
-        /// <param name="subMean">Subtract mean from each value before normalizing.</param>
-        /// <param name="useStdDev">Normalize by standard deviation rather than L2 norm.</param>
-        /// <param name="scale">Scale features by this value.</param>
-        public static IDataTransform CreateGlobalContrastNormalizer(IHostEnvironment env,
-            IDataView input,
-            string name,
-            string source = null,
-            bool subMean = Defaults.GcnSubMean,
-            bool useStdDev = Defaults.UseStdDev,
-            Float scale = Defaults.Scale)
-        {
-            var args = new GcnArguments()
-            {
-                Column = new[] { new GcnColumn(){
-                        Source = source ?? name,
-                        Name = name
-                    }
-                },
-                SubMean = subMean,
-                UseStdDev = useStdDev,
-                Scale = scale
-            };
-            return new LpNormNormalizerTransform(env, args, input);
+        private static (string input, string output)[] GetColumnPairs(ColumnInfoBase[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
+        }
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
+        {
+            var inType = inputSchema.GetColumnType(srcCol);
+            var reason = TestColumn(inType);
+            if (reason != null)
+                throw Host.ExceptParam(nameof(inputSchema), reason);
```

Comment:
>ExceptParam [](start = 27, length = 11)

please use `ExceptSchemaMismatch`

Reply:
I give you permission to change the message, but let's be more consistent in schema checking, and for that reason let's use `ExceptSchemaMismatch` whenever possible

---
In reply to: [232436987](https://github.com/dotnet/machinelearning/pull/1582#discussion_r232436987) [](ancestors = 232436987)

Reply:
I still don't see a schema mismatch error

---
In reply to: [232437023](https://github.com/dotnet/machinelearning/pull/1582#discussion_r232437023) [](ancestors = 232437023,232436987)